.github/scripts/transform_dbt_test_results.py

```diff
@@ -929,6 +954,104 @@ def create_list(
+
+    @classmethod
+    def create_list(
+        cls,
+        test_categories: typing.List[TestCategory],
+        run_results_filepath: str,
+    ) -> typing.List["TestRunFailingRowMetadata"]:
+        """Generate a list of TestRunFailingRowMetadata object from a list of
+        TestCategory objects representing the categories in the run and a
+        filepath to a run_results.json file."""
+        run_id = get_run_id_from_run_results(run_results_filepath)
+        run_date = get_run_date_from_run_results(run_results_filepath)
+        run_year = run_date[:4]
+
+        def value_to_list(value):
+            """Tiny helper function to convert column values to lists."""
+            if value is None or type(value) is list:
+                return value
+            return [value]
+
+        return [
+            TestRunFailingRowMetadata(
+                run_id=run_id,
+                run_year=run_year,
+                test_name=township_result.name,
+                table_name=township_result.table_name,
+                column_name=township_result.column_name,
+                category=test_category.category,
+                description=township_result.description,
+                township_code=township_result.township_code,
+                parid=failing_row.get(PARID_FIELD),
+                taxyr=failing_row.get(TAXYR_FIELD),
+                card=value_to_list(failing_row.get(CARD_FIELD)),
+                lline=value_to_list(failing_row.get(LAND_LINE_FIELD)),
+                class_=value_to_list(failing_row.get(CLASS_FIELD)),
+                who=value_to_list(failing_row.get(WHO_FIELD)),
+                wen=value_to_list(failing_row.get(WEN_FIELD)),
+                problematic_fields={
+                    key: val
+                    for key, val in failing_row.items()
+                    # Use possible_fixed_fieldnames to avoid having to
+                    # recompute the exact fixed fieldnames on every iteration
+                    # (we could also solve this by expanding out the list
+                    # comprehension, but for now this is easier)
+                    if key not in test_category.possible_fixed_fieldnames
+                },
+            )
+            for test_category in test_categories
+            for test_result in test_category.test_results
+            for township_result in test_result.split_by_township()
+            for failing_row in township_result.failing_rows
+            if township_result.status == Status.FAIL
+        ]
+
+    def to_dict(self) -> typing.Dict:
+        output_data = dataclasses.asdict(self)
+        # Serialize the "class" attribute to a more human-friendly name
+        output_data["class"] = output_data.pop("class_")
+        # Dump the problematic fields to string, since parquet can't handle
+        # the notion of an untyped JSON object
+        output_data["problematic_fields"] = json.dumps(
+            output_data["problematic_fields"]
+        )
```

Comment:
I also don't love that this field needs to be serialized to a string, but as far as I know parquet doesn't support untyped JSON columns.